src/TestGrains/ImplicitSubscription_NonTransientError_RecoverableStream_CollectorGrain.cs

```diff
@@ -0,0 +1,104 @@
+﻿
+using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.Streams;
+using TestGrainInterfaces;
+using UnitTests.Grains;
+
+namespace TestGrains
+{
+    [ImplicitStreamSubscription(StreamNamespace)]
+    public class ImplicitSubscription_NonTransientError_RecoverableStream_CollectorGrain : Grain<StreamCheckpoint<int>>, IGeneratedEventCollectorGrain
+    {
+        public const string StreamNamespace = "NonTransientError_RecoverableStream";
+     
+        // grain instance state
+        private Logger logger;
+        private IAsyncStream<GeneratedEvent> stream;
+
+        private class FaultsState
+        {
+            public bool FaultCleared { get; set; }
+        }
+        private static readonly ConcurrentDictionary<Guid, FaultsState> FaultInjectionTracker = new ConcurrentDictionary<Guid, FaultsState>();
```

Comment:
its a grain, right? So no need in `ConcurrentDictionary`. Regular `Dictionary` will be enough.